File: factorial.c
#include<stdio.h>

int factorial(int n);

int factorial(int n){
	if(n==1)
		return 1;
	else
		return n*factorial(n-1);
}
int main(){
	int n;
	printf("Enter the number: ");
	scanf("%d",&n);
	printf("Factorial: %d",factorial(n));
	return 0;
}

File: Factorial.java
import java.util.Scanner;

class Factorial{
	public static void main(String args[]){
		Scanner scan = new Scanner(System.in);
		System.out.print("Enter a number : ");
		int c =scan.nextInt();
		System.out.println("Factorial of "+c+" = "+fact(c));
	}
	static int fact(int n){
		if(n==1)
			return 1;
		else
			return(n*fact(n-1));
	}
}